packages/react-test-renderer/src/ReactShallowRenderer.js

```diff
@@ -619,6 +677,10 @@ class ReactShallowRenderer {
             ReactCurrentDispatcher.current = prevDispatcher;
           }
           this._finishHooks(element, context);
+          if (this._options.callEffects) {
+            this._callEffects(true);
```

Comment:
This basically just toggles it on and off—is there anyway to condense it into one func?

Reply:
@ssunday that's not what it does, first it runs the layout effects, then the effects.

Reply:
Sorry, I misread some of the logic. Trying add comments to something very good. So eager to get this going!

Reply:
Same!